### PR TITLE
Laboratory work #3, Valeria Bogdanova - 23FPL1

### PR DESCRIPTION
**Pull request by lerabogdanova**

### CODE REVIEW

File: 2024-2-level-labs
Comment (artyomtugaryov) [inline 2024-2-level-labs:1]: How did you do this? Revert this change!

File: lab_3_ann_retriever/main.py [hunk 1]
```diff
@@ -104,6 +105,11 @@ def tokenize(self, text: str) -> list[str] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not text or not isinstance(text, str):
+            return None
+
+        tokens = ''.join([symb for symb in text if symb.isalpha() or symb == " "]).split()
+        return Tokenizer._remove_stop_words(self, tokens)
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:112]:
```suggestion
        return self._remove_stop_words(tokens)
```

File: lab_3_ann_retriever/main.py [hunk 2]
```diff
@@ -117,6 +123,14 @@ def tokenize_documents(self, documents: list[str]) -> list[list[str]] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not documents or not isinstance(documents, list) \
+            or not all(isinstance(doc, str) for doc in documents):
+            return None
+
+        tok_documents = []
+        for doc in documents:
+            tok_documents.append(Tokenizer.tokenize(self, doc))
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:132]: the same

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Is it really for 6? If so, try for 8

File: lab_3_ann_retriever/main.py (hunk 2 above)
Comment (lerabogdanova) [reply lab_3_ann_retriever/main.py:132]: Done

File: lab_3_ann_retriever/main.py (hunk 1 above)
Comment (lerabogdanova) [reply lab_3_ann_retriever/main.py:112]: Done

File: 2024-2-level-labs
Comment (artyomtugaryov) [inline 2024-2-level-labs:1]: What is it? revert

File: lab_3_ann_retriever/main.py [hunk 3]
```diff
@@ -158,6 +199,23 @@ def build(self) -> bool:
...
+        self._vocabulary.sort()
+        for token in self._vocabulary:
+            self._token2ind[token] = self._vocabulary.index(token)
+        self._idf_values = calculate_idf(self._vocabulary, self._corpus)
+
+        if not self._idf_values or not self._vocabulary or not self._token2ind \
+                or not all(token for token in self._vocabulary) \
+                or calculate_idf(self._vocabulary, self._corpus) is None:
+            return False
+        else:
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:215]: remove `else`

File: lab_3_ann_retriever/main.py [hunk 4]
```diff
@@ -118,6 +139,16 @@ def tokenize_documents(self, documents: list[str]) -> list[list[str]] | None:
         In case of corrupt input arguments, None is returned.
         """
 
+        if not documents or not isinstance(documents, list) \
+            or not all(isinstance(doc, str) for doc in documents):
+            return None
+
+        tok_documents = []
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:146]: list comprehension

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_ann_retriever/main.py (hunk 3 above)
Comment (lerabogdanova) [reply lab_3_ann_retriever/main.py:213]: Done

**Pull request closed without merging**